### PR TITLE
Realloc validator and stake lists

### PR DESCRIPTION
**Pull request by aankor**

### CODE REVIEW

File: programs/marinade-finance/src/state/validator_system.rs [hunk 1]
```diff
@@ -73,6 +73,39 @@ impl ValidatorRecord {
...
+        }
+        if buf[0..8] != Self::DISCRIMINATOR {
+            return err!(MarinadeError::InvalidValidatorListDiscriminator);
+        }
+        *buf = &buf[8..];
+        Ok(Self {})
+    }
+}
+
+impl AccountSerialize for ValidatorList {}
```
Comment (luciotato) [inline programs/marinade-finance/src/state/validator_system.rs:101]: @aankor just to confirm, when you do this:
`impl AccountSerialize for ValidatorList {}`
Then Anchor does not try to write the account data at the end of the instruction, correct?
If that's so can you please add a comment saying:
"By implementing an empty AccountSerialize, Anchor will not try to write the account data at the end of the instruction. The account data is written explicitly in the code"

Comment (aankor) [reply programs/marinade-finance/src/state/validator_system.rs:105]: Added comment

**Review by ochaloup — APPROVED** [review]

File: programs/marinade-finance/src/error.rs [hunk 2]
```diff
@@ -265,4 +265,7 @@ pub enum MarinadeError {
     /// when the remainder after a withdraw stake account is less than min_stake
     #[msg("Stake account remainder too low")]
     StakeAccountRemainderTooLow, // 6085 0x17c5
+
+    #[msg("Capacity of the list must be not less than it's current size")]
+    ShrinkingListWithDeletingContents, // 6086 0x17c6
```
[the remark this answers is not in the document]
Comment (luciotato) [reply programs/marinade-finance/src/error.rs:270]: I think the plural "contents" is correct

File: programs/marinade-finance/src/instructions/admin/realloc_stake_list.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        realloc::payer = rent_funds,
+        realloc::zero = false,
+    )]
+    pub stake_list: Account<'info, StakeList>,
+
+    #[account(
+        mut,
+        owner = system_program::ID,
+    )]
+    pub rent_funds: Signer<'info>,
```
Comment (luciotato) [inline programs/marinade-finance/src/instructions/admin/realloc_stake_list.rs:28]: any reason why not use the standard name "rent_payer" ?

**Review by luciotato — COMMENTED** [review]
LGTM

**Pull request merged**